Xamarin.PropertyEditing.Mac/Controls/Custom/CommandMenuItem.cs

```diff
@@ -0,0 +1,34 @@
+﻿using System;
+using System.Windows.Input;
+using AppKit;
+
+namespace Xamarin.PropertyEditing.Mac
+{
+	public class CommandMenuItem : NSMenuItem
```

Comment:
This doesn't track `ICommand.CanExecuteChanged`

Reply:
Unless I'm missing something, as we have a reference to the actual ClearValueCommand, that will fire automagically, inside ValidateMenuItem, when command.CanExecute is called and under the hood CanClearValue () is called to make sure it can. The behaviour on both Windows and Mac, from the UI perspective, is identical. They enable and disable as they are supposed to.

Reply:
That only works assuming that the can execute state doesn't change _while_ the item is being displayed. Presumably, they are only calling validate when the menu opens.

Reply:
Yes it is only called when you try to open the menu. 
IIRC Execute, calls CanExecute again, just before actually executing the required call. At least that is how I've written ICommands in the past and gets around that very scenario.

Reply:
But in this case, how is the VM for that particular property likely to change, without user interaction?

Reply:
Maybe there's some button we hit in the menu itself that doesn't end up closing it. Maybe some async update comes in after the menu opens.

We shouldn't cut corners just because we can't think of a scenario where this case would occur. If nothing else we should properly handle this contract to be defensive. Some feature we implement a year from now could depend on this and because Windows handles the contract properly it works, but then we have a bug in Mac we have to find and fix.

Reply:
I'm not looking to cut corners Eric, but trying to establish real world situations. 
I agree that the contract should be upheld in case of future dependency.